fixtures/syntax.js

```diff
@@ -4,15 +4,29 @@ import * as namespaceImport from 'bar'
 const dynamicImport = import('baz')
 
 class Foo {
+  *methodGenerator(seed) {
+    yield seed
+  }
   static bar = 'abc'
   baz = (x, y) => x({...y})
 }
 
+function* generator() {
+  const input = yield 'using last argument passed to next()'
+  yield* new Foo().methodGenerator(input)
+}
+
+async function asyncFunction() {
+  return await null
+}
```

Comment:
Makes a lot of sense, I was also thinking that might be better. One suggestion I have is to keep `yield*` as a sanity check since it's a distinct operator from `yield`, like with the spread syntax where we check both r-value spread `val = {...obj}` and  l-value spread `{val, ...rest} = obj`. This is a bare-bones version I'm thinking of:
```
function* infiniteGenerator() {
  yield 1
  yield* infiniteGenerator()
}
```